sql/core/src/test/resources/sql-tests/results/group-by-filter.sql.out

```diff
@@ -0,0 +1,569 @@
+10	200.0
+100	400.0
+20	300.0
+30	NULL
+70	150.0
+NULL	400.0
+
+
+-- !query 31
+SELECT a AS k, COUNT(b) FILTER (WHERE NOT b < 0) FROM testData GROUP BY k HAVING k > 1
+-- !query 31 schema
+struct<k:int,count(b):bigint>
+-- !query 31 output
+2	2
+3	2
+
+
+-- !query 32
+SELECT dept_id AS k, AVG(salary) FILTER (WHERE NOT hiredate <= date "2005-01-01")
+FROM emp GROUP BY k HAVING k < 70
+-- !query 32 schema
+struct<k:int,avg(salary):double>
+-- !query 32 output
+10	NULL
+20	NULL
+30	NULL
+
+
+-- !query 33
+SELECT dept_id AS k, AVG(salary) FILTER (WHERE NOT hiredate <= to_date("2005-01-01"))
+FROM emp GROUP BY k HAVING k < 70
+-- !query 33 schema
+struct<k:int,avg(salary):double>
+-- !query 33 output
+10	NULL
+20	NULL
+30	NULL
+
+
+-- !query 34
+SELECT dept_id AS k, AVG(salary) FILTER (WHERE NOT hiredate <= to_timestamp("2005-01-01 00:00:00"))
+FROM emp GROUP BY k HAVING k < 70
+-- !query 34 schema
+struct<k:int,avg(salary):double>
+-- !query 34 output
+10	NULL
+20	NULL
+30	NULL
+
+
+-- !query 35
+SELECT COUNT(b) FILTER (WHERE a > 0) AS k FROM testData GROUP BY k
+-- !query 35 schema
+struct<>
+-- !query 35 output
+org.apache.spark.sql.AnalysisException
+aggregate functions are not allowed in GROUP BY, but found count(testdata.`b`);
+
+
+-- !query 36
+SELECT a AS k, COUNT(b) FILTER (WHERE b > 0) FROM testData GROUP BY k
+-- !query 36 schema
+struct<k:int,count(b):bigint>
+-- !query 36 output
+1	2
+2	2
+3	2
+NULL	1
+
+
+-- !query 37
+SELECT a, COUNT(1) FILTER (WHERE b > 1) FROM testData WHERE false GROUP BY a
+-- !query 37 schema
+struct<a:int,count(1):bigint>
+-- !query 37 output
+
+
+
+-- !query 38
+SELECT COUNT(1) FILTER (WHERE b = 2) FROM testData WHERE false
+-- !query 38 schema
+struct<count(1):bigint>
+-- !query 38 output
+0
+
+
+-- !query 39
+SELECT 1 FROM (SELECT COUNT(1) FILTER (WHERE a >= 3 OR b <= 1) FROM testData WHERE false) t
+-- !query 39 schema
+struct<1:int>
+-- !query 39 output
+1
+
+
+-- !query 40
+SELECT emp.dept_id,
+       avg(salary),
+       avg(salary) FILTER (WHERE id > (SELECT 200))
+FROM emp
+GROUP BY dept_id
+-- !query 40 schema
+struct<dept_id:int,avg(salary):double,avg(salary):double>
+-- !query 40 output
+10	133.33333333333334	NULL
+100	400.0	400.0
+20	300.0	300.0
+30	400.0	400.0
+70	150.0	150.0
+NULL	400.0	400.0
+
+
+-- !query 41
+SELECT emp.dept_id,
+       avg(salary),
+       avg(salary) FILTER (WHERE emp.dept_id = (SELECT dept_id FROM dept LIMIT 1))
+FROM emp
+GROUP BY dept_id
+-- !query 41 schema
+struct<dept_id:int,avg(salary):double,avg(salary):double>
+-- !query 41 output
+10	133.33333333333334	133.33333333333334
+100	400.0	NULL
+20	300.0	NULL
+30	400.0	NULL
+70	150.0	NULL
+NULL	400.0	NULL
+
+
+-- !query 42
+SELECT emp.dept_id,
+       avg(salary),
+       avg(salary) FILTER (WHERE EXISTS (SELECT state
+               FROM dept
+               WHERE dept.dept_id = emp.dept_id))
+FROM emp
+GROUP BY dept_id
+-- !query 42 schema
+struct<>
+-- !query 42 output
+org.apache.spark.sql.AnalysisException
```

Comment:
Please file jira for unsupported subquery cases and leave jira IDs here?

Reply:
OK